admin_manual/enterprise/external_storage/s3_swift_as_primary_object_store_configuration.rst

```diff
@@ -1,159 +1,162 @@
+
+    3. The current implementation is incompatible with any app that uses direct file I/O (input/output) as it circumvents the ownCloud virtual filesystem. Two excellent examples are:
+
+       A. **The Encryption app:** It fetches critical files in addition to any requested file, which results in significant overhead.
+       B. **The Gallery app:** It stores thumbnails directly in the filesystem.
 
 Configuration
 -------------
 
-Look in ``config.sample.php`` for a example configurations. Copy the 
-relevant part to your ``config.php`` file. Any object store needs to implement
-``\\OCP\\Files\\ObjectStore\\IObjectStore`` and can be passed parameters in the
-constructor with the ``arguments`` key:
+Look in ``config.sample.php`` for example configurations.
+Copy the relevant part to your ``config.php`` file.
+Any object store needs to implement ``\\OCP\\Files\\ObjectStore\\IObjectStore``, and can be passed parameters in the constructor with the ``arguments`` key, as in the following example:
 
-::
+.. code-block:: php
 
-    'objectstore' => [
-        'class' => 'Implementation\\Of\\OCP\\Files\\ObjectStore\\IObjectStore',
-        'arguments' => [
-            ...
+    $CONFIG = [
+        'objectstore' => [
+            'class' => 'Implementation\\Of\\OCP\\Files\\ObjectStore\\IObjectStore',
+            'arguments' => [
+                ...
+            ],
         ],
     ],
 
 Amazon S3
 ~~~~~~~~~
 
-The S3 backend mounts a bucket of the Amazon S3 object store
-into the virtual filesystem. The class to be used is ``OCA\ObjectStore\S3``:
-
-::
-
-  'objectstore' => [
-      'class' => 'OCA\ObjectStore\S3',
-      'arguments' => [
-          // replace with your bucket
-          'bucket' => 'owncloud',
-          'autocreate' => true,
-          // uncomment to enable server side encryption
-          //'serversideencryption' => 'AES256',
-          'options' => [
-              // version and region are required
-              'version' => '2006-03-01',
-              // change to your region
-              'region'  => 'eu-central-1',
-              'credentials' => [
-                  // replace key and secret with your credentials
-                  'key' => 'EJ39ITYZEUH5BGWDRUFY',
-                  'secret' => 'M5MrXTRjkyMaxXPe2FRXMTfTfbKEnZCu+7uRTVSj',
-              ],
-          ],
-      ],
-  ],
+The S3 backend mounts a bucket of the Amazon S3 object store into the virtual filesystem.
+The class to be used is ``OCA\Files_Primary_S3\S3Storage``, as in the following example:
+
+.. code-block:: php
+
+   <?php
+
+    $CONFIG = [
+        'objectstore' => [
+            'class' => 'OCA\Files_Primary_S3\S3Storage',
+            'arguments' => [
+                // replace with your bucket
+                'bucket' => 'owncloud',
+                'autocreate' => true,
```

Comment:
This no longer exists ref: `https://github.com/owncloud/files_primary_s3/pull/93`

Reply:
@patrickjahns, is this because of an update to the aws-sdk-php?